src/main/java/com/mraof/minestuck/entity/animation/MobAnimation.java

```diff
@@ -0,0 +1,93 @@
+package com.mraof.minestuck.entity.animation;
+
+import net.minecraft.nbt.CompoundTag;
+
+/**
+ * Keeps track of a given animation. Can store stages of the animation in which to perform additional actions/events
+ *
+ * @param actions         category of the animation
+ * @param animationLength last frame of the animation
+ * @param freezeMovement  whether the entity can move voluntarily during an animation, handled both in Goals and is updated upon the change of animation in AnimatedPathfinderMob
+ * @param freezeSight     whether the entity has their facing direction locked, handled by Goals
+ */
+public record MobAnimation(Actions actions, int animationLength, boolean freezeMovement, boolean freezeSight)
+{
+	public static final int LOOPING_ANIMATION = -1; //when set to -1, it will continue looping until another animation overrides it
+	public static final MobAnimation.Actions IDLE_ACTION = Actions.IDLE;
+	public static final MobAnimation DEFAULT_IDLE_ANIMATION = new MobAnimation(IDLE_ACTION, LOOPING_ANIMATION, false, false);
+	
+	public Actions getAction()
+	{
+		return actions;
+	}
+	
+	public int getAnimationLength()
+	{
+		return animationLength;
+	}
+	
+	public boolean isLoopingAnimation()
+	{
+		return animationLength == LOOPING_ANIMATION;
+	}
+	
+	public boolean freezesMovement()
+	{
+		return freezeMovement;
+	}
+	
+	public boolean freezesSight()
+	{
+		return freezeSight;
+	}
```

Comment:
Since the class is now a record, some of these getters will already exist as part of the record.